src/Omega_h_adios2.cpp

```diff
@@ -86,55 +103,59 @@ static void read_array(adios2::IO &io, adios2::Engine &reader,
                Mesh* mesh, Read<T> &array, std::string &name)
 {
   long unsigned int rank = mesh->comm()->rank();
+  
+  auto var = io.InquireVariable(name);
+  std::vector<size_t> shape = var.Count();
+  assert(shape.size() == 2);
+  size_t Nx = shape[0];
+  size_t Ny = shape[1];
+  HostWrite<T> array_(Nx*Ny);
 
-  std::string subname = name + "_size";
-  size_t Nx=1;
-  read_value(io, reader, mesh->comm(), &Nx, subname);
-
-  HostWrite<T> array_(Nx);
-
-  adios2::Variable<T> bpData = io.InquireVariable<T>(name+"_data");
+  adios2::Variable<T> bpData = io.InquireVariable<T>(name);
   if (bpData) // means found
   {
     std::vector<T> myData;
 
     // read only the chunk corresponding to this rank
-    bpData.SetSelection({{Nx * rank}, {Nx}});
+    bpData.SetSelection({{Nx * rank,0}, {Nx,Ny}});
     reader.Get(bpData, myData, adios2::Mode::Sync);
-    for (LO x=0; x<(LO)Nx; ++x)
-      array_.set(x, myData[x]);
-
+    for (LO x=0; x<(LO)(Nx*Ny); ++x)
+        array_.set(x, myData[x]);
     array=Read<T>(array_.write());
   }
 }
 
-
 static void write_down(adios2::IO &io, adios2::Engine &writer, Mesh* mesh, int d, std::string pref)
 {
   auto down = mesh->ask_down(d, d - 1);
-  std::string name = pref+"down.ab2b_" + std::to_string(d);
-  write_array(io, writer, mesh, down.ab2b, name);
+  int ncomp = d+1;  // only valid for 2D triangle, and 3D tetrahedron.
```

Comment:
Please add an assert/fail here that the 'family' of the mesh is simplex (i.e., https://github.com/SCOREC/omega_h/blob/09a51716b37c630ab04be85e50ff2f8014383f5d/src/Omega_h_mesh.hpp#L54 and https://github.com/SCOREC/omega_h/blob/09a51716b37c630ab04be85e50ff2f8014383f5d/src/Omega_h_defines.hpp#L70).

Reply:
Do we have a function that returns the appropriate number of downward adjacency for each of these types?

Reply:
Yes.  https://github.com/SCOREC/omega_h/blob/ed19d260411804730605cb0b91d46d30bc919b89/src/Omega_h_simplex.hpp#L313